src/core/labeling/qgspallabeling.cpp

```diff
@@ -2364,7 +2386,7 @@ std::unique_ptr<QgsLabelFeature> QgsPalLayerSettings::registerFeatureWithDetails
 
   //data defined quadrant offset?
   bool ddFixedQuad = false;
-  Qgis::LabelQuadrantPosition quadOff = quadOffset;
+  Qgis::LabelQuadrantPosition quadOff = pointSettings.quadrant();
   if ( mDataDefinedProperties.isActive( QgsPalLayerSettings::Property::OffsetQuad ) )
```

Comment:
Why aren't you moving this data defined handling into QgsLabelPointSettings::updateDataDefinedProperties()? Same goes for other properties that were moved into QgsLabelPointSettings.

Reply:
Just to avoid any unforeseen issues. Ideally yes, they'd be in here, but I wanted to avoid messing with the old logic as much as possible

Reply:
I've added a TODO for this